aten/src/ATen/core/MT19937RNGEngine.h

```diff
@@ -118,7 +118,7 @@ class mt19937_engine {
     return data_;
   }
 
-  inline void set_data(mt19937_data_pod data) {
```

Comment:
mt19937 is only copyable? no move ctor?

Reply:
https://github.com/pytorch/pytorch/blob/0a07488ed2c47765e337e290bd138c0e6e459cbd/aten/src/ATen/core/MT19937RNGEngine.h#L102 Ah, I see, it should be trivially copyable, but it has a potentially large std::array.

Reply:
POD data, their is no gain to move it.

Reply:
https://clang.llvm.org/extra/clang-tidy/checks/modernize/pass-by-value.html can you please explain the difference between pass-by-reference+copy vs pass-by-value+move?

Reply:
@malfet if the param is given as an rvalue, then no copy is performed. This answer gives a great explanation of the difference between each approach: https://stackoverflow.com/a/51706522/2444240